cmd/app.go

```diff
@@ -13,13 +17,87 @@ func (a *WebScan) InitAppCommand() {
 	a.AppCmd = &cobra.Command{
 		Use:   "app",
 		Short: "Perform various application scans",
-		Long:  `Perform various application scans such as fingerprinting and enumeration`,
+		Long:  `Perform various application scans such as detect and enumeration`,
 	}
 
 	a.RootCmd.AddCommand(a.AppCmd)
+	a.initDetectCommand()
 	a.initEnumerateCommand()
 }
 
+func (a *WebScan) initDetectCommand() {
+	detectCmd := &cobra.Command{
+		Use:   "detect",
+		Short: "Perform a detection scan against a target",
+		Long: `Perform a detection scan against a target using specified types.
+		
+The detection command identifies the type of web application running on the target URL.
+It supports detecting different resource types including API applications (FastAPI, Swagger, gRPC, GraphQL, K8s), and 
+cloud buckets (AWSS3, AzureBlob). The command accepts a list of modules to run
+for the specified resource type.`,
+		Run: func(cmd *cobra.Command, args []string) {
+			targets, err := cmd.Flags().GetStringSlice("targets")
+			if err != nil {
+				a.OutputSignal.AddError(err)
+				return
+			}
+			resourceType, err := cmd.Flags().GetString("resourcetype")
+			if err != nil {
+				a.OutputSignal.AddError(err)
+				return
+			}
+			resourceTypeEnum, err := validateDetectResourceType(resourceType)
+			if err != nil {
+				a.OutputSignal.AddError(err)
+				return
+			}
+			modules, err := cmd.Flags().GetStringSlice("modules")
+			if err != nil {
+				a.OutputSignal.AddError(err)
+				return
+			}
+			moduleEnums, err := validateDetectResourseModuleSelection(*resourceTypeEnum, modules)
+			if err != nil {
+				a.OutputSignal.AddError(err)
+				return
+			}
+			timeout, err := cmd.Flags().GetInt("timeout")
+			if err != nil {
+				a.OutputSignal.AddError(err)
+				return
+			}
+			successfulOnly, err := cmd.Flags().GetBool("successfulonly")
+			if err != nil {
+				a.OutputSignal.AddError(err)
+				return
+			}
+			config, err := newDetectConfig(targets, *resourceTypeEnum, moduleEnums, timeout, successfulOnly)
+			if err != nil {
+				a.OutputSignal.AddError(err)
+				return
+			}
+
+			engine := detect.NewEngine(config)
+			report, err := engine.Launch(cmd.Context())
+			if err != nil {
+				a.OutputSignal.AddError(err)
+			}
+			a.OutputSignal.Content = report
+		},
+	}
+
+	detectCmd.Flags().StringSlice("targets", []string{}, "URL target to perform detect against")
+	detectCmd.Flags().String("resourcetype", "", "Resource type to detect")
+	detectCmd.Flags().StringSlice("modules", []string{}, "Modules to run")
```

Comment:
Same as above. What are valid modules? Also, it's not clear what the difference between a resource type and a module is or how they interact.